diff --git a/dlang.org/posix.mak b/dlang.org/posix.mak
index dde619a1..24745b71 100644
--- a/dlang.org/posix.mak
+++ b/dlang.org/posix.mak
@@ -176,7 +176,7 @@ G=$(GENERATED)
 # Last released versions
 DMD_LATEST_DIR=$G/dmd-${LATEST}
 DMD_LATEST=$(DMD_LATEST_DIR)/generated/$(OS)/release/$(MODEL)/dmd
-DRUNTIME_LATEST_DIR=$G/druntime-${LATEST}
+DRUNTIME_LATEST_DIR=$G/dmd-${LATEST}/druntime
 PHOBOS_LATEST_DIR=$G/phobos-${LATEST}
 
 # stable dub and dmd versions used to build dpl-docs
@@ -431,10 +431,9 @@ ${GENERATED}/modlist-prerelease.ddoc : tools/modlist.d ${STABLE_DMD} $(DRUNTIME_
 		$(addprefix --dump , object std etc core dmd rt core.internal.array core.internal.util) >$@
 
 # Run "make -j rebase" for rebasing all dox in parallel!
-rebase: rebase-dlang rebase-dmd rebase-druntime rebase-phobos
+rebase: rebase-dlang rebase-dmd rebase-phobos
 rebase-dlang: ; $(call REBASE,dlang.org)
 rebase-dmd: ; cd $(DMD_DIR) && $(call REBASE,dmd)
-rebase-druntime: ; cd $(DRUNTIME_DIR) && $(call REBASE,druntime)
 rebase-phobos: ; cd $(PHOBOS_DIR) && $(call REBASE,phobos)
 
 clean:
@@ -575,12 +574,16 @@ $G/dblog_latest.ddoc: $G/dblog_latest.xml $(STABLE_DMD) tools/ddoc_xml_extractor
 # Git rules
 ################################################################################
 
+# Druntime is in the DMD repository.
+${DRUNTIME_DIR}: ${DMD_DIR}
+${DRUNTIME_LATEST_DIR}: ${DMD_LATEST_DIR}
+
 # Clone snapshots of the latest official release of all main D repositories
 $G/%-${LATEST} :
 	git clone -b v${LATEST} --depth=1 ${GIT_HOME}/$(notdir $*) $@
 
 # Clone all main D repositories
-${DMD_DIR} ${DRUNTIME_DIR} ${PHOBOS_DIR} ${TOOLS_DIR} ${INSTALLER_DIR} ${DUB_DIR}:
+${DMD_DIR} ${PHOBOS_DIR} ${TOOLS_DIR} ${INSTALLER_DIR} ${DUB_DIR}:
 	git clone ${GIT_HOME}/$(notdir $(@F)) $@
 
 ${DMD_DIR}/VERSION : ${DMD_DIR}
diff --git a/dlang.org/posix.mak b/dlang.org/posix.mak
index 24745b71..f1239280 100644
--- a/dlang.org/posix.mak
+++ b/dlang.org/posix.mak
@@ -597,7 +597,7 @@ $(DMD) : ${DMD_DIR}
 
 $(DMD_LATEST) : ${DMD_LATEST_DIR}
 	${MAKE} --directory=${DMD_LATEST_DIR}/src -f posix.mak AUTO_BOOTSTRAP=1
-	sed -i -e "s|../druntime/import |../druntime-${LATEST}/import |" -e "s|../phobos |../phobos-${LATEST} |" $@.conf
+	sed -i -e "s|../druntime/import |../../dmd-${LATEST}/druntime/import |" -e "s|../phobos |../phobos-${LATEST} |" $@.conf
 
 dmd-prerelease : $(STD_DDOC_PRERELEASE) druntime-target $G/changelog/next-version
 	$(MAKE) AUTO_BOOTSTRAP=1 --directory=$(DMD_DIR) -f posix.mak html $(DDOC_VARS_PRERELEASE_HTML)
diff --git a/dlang.org/posix.mak b/dlang.org/posix.mak
index f1239280..3575a812 100644
--- a/dlang.org/posix.mak
+++ b/dlang.org/posix.mak
@@ -414,7 +414,7 @@ ${GENERATED}/${LATEST}.ddoc :
 
 ${GENERATED}/modlist-${LATEST}.ddoc : tools/modlist.d ${STABLE_DMD} $(DRUNTIME_LATEST_DIR) $(PHOBOS_LATEST_DIR) $(DMD_LATEST_DIR)
 	mkdir -p $(dir $@)
-	$(STABLE_RDMD) $< $(DRUNTIME_LATEST_DIR)/src $(PHOBOS_LATEST_DIR) $(DMD_LATEST_DIR)/src $(MOD_EXCLUDES_LATEST) \
+	$(STABLE_RDMD) $< $(DRUNTIME_LATEST_DIR)/src $(PHOBOS_LATEST_DIR) $(DMD_LATEST_DIR)/compiler/src $(MOD_EXCLUDES_LATEST) \
 		$(addprefix --internal=, dmd rt core.internal) \
 		$(addprefix --dump , object std etc core dmd rt core.internal.array core.internal.util) >$@
 
diff --git a/dlang.org/posix.mak b/dlang.org/posix.mak
index 3575a812..1eede5ba 100644
--- a/dlang.org/posix.mak
+++ b/dlang.org/posix.mak
@@ -155,7 +155,7 @@ DUB_DIR=../dub
 
 # Auto-cloning missing directories
 $(shell [ ! -d $(DMD_DIR) ] && git clone --depth=1 ${GIT_HOME}/dmd $(DMD_DIR))
-include $(DMD_DIR)/src/osmodel.mak
+include $(DMD_DIR)/compiler/src/osmodel.mak
 
 # External binaries
 DMD=$(DMD_DIR)/generated/$(OS)/release/$(MODEL)/dmd
@@ -593,10 +593,10 @@ ${DMD_DIR}/VERSION : ${DMD_DIR}
 ################################################################################
 
 $(DMD) : ${DMD_DIR}
-	${MAKE} --directory=${DMD_DIR}/src -f posix.mak AUTO_BOOTSTRAP=1
+	${MAKE} --directory=${DMD_DIR}/compiler/src -f posix.mak AUTO_BOOTSTRAP=1
 
 $(DMD_LATEST) : ${DMD_LATEST_DIR}
-	${MAKE} --directory=${DMD_LATEST_DIR}/src -f posix.mak AUTO_BOOTSTRAP=1
+	${MAKE} --directory=${DMD_LATEST_DIR}/compiler/src -f posix.mak AUTO_BOOTSTRAP=1
 	sed -i -e "s|../druntime/import |../../dmd-${LATEST}/druntime/import |" -e "s|../phobos |../phobos-${LATEST} |" $@.conf
 
 dmd-prerelease : $(STD_DDOC_PRERELEASE) druntime-target $G/changelog/next-version
@@ -719,13 +719,13 @@ $W/library-prerelease/.htaccess : dpl_prerelease_htaccess
 
 $G/docs-latest.json : ${DMD_LATEST} ${DMD_LATEST_DIR} \
 			${DRUNTIME_LATEST_DIR} | dpl-docs
-	find ${DMD_LATEST_DIR}/src -name '*.d' -o -name '*.di' | sort -r | \
+	find ${DMD_LATEST_DIR}/compiler/src -name '*.d' -o -name '*.di' | sort -r | \
 		gawk '!n[gensub(/\.di?$$/, "", 1)]++' > $G/.latest-files.txt
 	find ${DRUNTIME_LATEST_DIR}/src -name '*.d' | \
 		sed -e /unittest.d/d -e /gcstub/d >> $G/.latest-files.txt
 	find ${PHOBOS_LATEST_DIR}/etc ${PHOBOS_LATEST_DIR}/std -name '*.d' | \
 		sed -e /unittest.d/d | sort >> $G/.latest-files.txt
-	${DMD_LATEST} -J$(DMD_LATEST_DIR)/src/dmd/res -J$(dir $(DMD_LATEST)) -c -o- -version=CoreDdoc \
+	${DMD_LATEST} -J$(DMD_LATEST_DIR)/compiler/src/dmd/res -J$(dir $(DMD_LATEST)) -c -o- -version=CoreDdoc \
 		-version=MARS -version=CoreDdoc -version=StdDdoc -Df$G/.latest-dummy.html \
 		-Xf$@ -I${PHOBOS_LATEST_DIR} @$G/.latest-files.txt
 	${DPL_DOCS} ${DPL_DOCS_FLAGS} filter $@ --min-protection=Protected \
@@ -733,7 +733,7 @@ $G/docs-latest.json : ${DMD_LATEST} ${DMD_LATEST_DIR} \
 	rm -f $G/.latest-files.txt $G/.latest-dummy.html
 
 $G/docs-prerelease.json : ${DMD} ${DMD_DIR} ${DRUNTIME_DIR} | dpl-docs
-	find ${DMD_DIR}/src -name '*.d' -o -name '*.di' | sort -r | \
+	find ${DMD_DIR}/compiler/src -name '*.d' -o -name '*.di' | sort -r | \
 		gawk '!n[gensub(/\.di?$$/, "", 1)]++' > $G/.prerelease-files.txt
 	find ${DRUNTIME_DIR}/src -name '*.d' | \
 		sed -e /unittest/d >> $G/.prerelease-files.txt
diff --git a/dlang.org/js/listanchors.js b/dlang.org/js/listanchors.js
index b7c113a3..34e7c4f5 100644
--- a/dlang.org/js/listanchors.js
+++ b/dlang.org/js/listanchors.js
@@ -66,45 +66,43 @@ function addVersionSelector() {
       return {
         name: e,
         url: "https://docarchives.dlang.io/v" + e + ".0/" + currentRoot + ddocModuleURL,
-        selected: false,
       };
   });
 
-  var rootURL = location.href.split(/\/(phobos|library|spec)(-prerelease)?/)[0]
   var onlineVersions;
   if (isSpec) {
     onlineVersions = [{
       name: "master",
-      url: rootURL + "/spec/" + ddocModuleURL,
+      url: "https://dlang.org/spec/" + ddocModuleURL,
     }];
   } else {
     onlineVersions = [{
       name: "master",
-      url: rootURL + "/phobos-prerelease/" + ddocModuleURL,
+      url: "https://dlang.org/phobos-prerelease/" + ddocModuleURL,
     },{
       name: "master (ddox)",
-      url: rootURL + "/library-prerelease/" + ddoxModuleURL,
+      url: "https://dlang.org/library-prerelease/" + ddoxModuleURL,
     },{
       name: "stable",
-      url: rootURL + "/phobos/" + ddocModuleURL,
+      url: "https://dlang.org/phobos/" + ddocModuleURL,
     },{
       name: "stable (ddox)",
-      url: rootURL + "/library/" + ddoxModuleURL,
+      url: "https://dlang.org/library/" + ddoxModuleURL,
     }];
   }
 
   // set the current URL as selected
   var currentURL = location.href.split(/[#?]/)[0];
-  onlineVersions.forEach(function(v, i) {
-    onlineVersions[i].selected = v.url === currentURL;
+  var versions = onlineVersions.concat(archivedVersions);
+  versions.forEach(function(v, i) {
+    versions[i].selected = v.url === currentURL;
   });
   // Don't show the option chooser if the page hasn't been recognized
   // For example, Ddox symbol pages are currently not supported
-  if (onlineVersions.filter(function(v){return v.selected}).length === 0)
+  if (versions.filter(function(v){return v.selected}).length === 0)
     return;
 
   // build select box of all versions and append to current DOM
-  var versions = onlineVersions.concat(archivedVersions);
   var options = versions.map(function(e, i){
     return "<option value='" + i + "'" + (e.selected ? "selected" : "") + ">" + e.name + "</option>";
   });
