diff --git a/tools/changed.d b/tools/changed.d
index 095eca9..e86626e 100755
--- a/tools/changed.d
+++ b/tools/changed.d
@@ -147,21 +147,33 @@ auto getIssues(string revRange)
         auto p = pipeProcess(cmd, Redirect.stdout);
         enforce(wait(p.pid) == 0, "Failed to execute '%(%s %)'.".format(cmd));
 
-        cmd = ["git", "-C", repo, "log", revRange];
-        p = pipeProcess(cmd, Redirect.stdout);
-        scope(exit) enforce(wait(p.pid) == 0, "Failed to execute '%(%s %)'.".format(cmd));
+        void runCmd(string[] cmd)
+        {
+            auto p = pipeProcess(cmd, Redirect.stdout);
+            scope(exit) enforce(wait(p.pid) == 0, "Failed to execute '%(%s %)'.".format(cmd));
 
-        foreach (line; p.stdout.byLine())
-        {
-            if (auto m = match(line.stripLeft, closedRE))
+            foreach (line; p.stdout.byLine())
             {
-                m.captures[1]
-                    .splitter(ctRegex!`[^\d]+`)
-                    .filter!(b => b.length)
-                    .map!(to!int)
-                    .copy(issues);
+                if (auto m = match(line.stripLeft, closedRE))
+                {
+                    m.captures[1]
+                        .splitter(ctRegex!`[^\d]+`)
+                        .filter!(b => b.length)
+                        .map!(to!int)
+                        .copy(issues);
+                }
             }
         }
+        if (repo.endsWith("/dmd"))
+        {
+            repo.buildPath("druntime").mkdirRecurse;
+            runCmd(["git", "-C", repo, "log", revRange, ":!druntime"]);
+            runCmd(["git", "-C", repo, "log", revRange, "druntime"]);
+        }
+        else if (repo.endsWith("/druntime"))
+            runCmd(["git", "-C", repo, "log", "v2.100.0..upstream/stable"]);
+        else
+            runCmd(["git", "-C", repo, "log", revRange]);
     }
     return issues.data.sort().release.uniq;
 }
diff --git a/tools/contributors.d b/tools/contributors.d
index d2c01df..4160e1f 100755
--- a/tools/contributors.d
+++ b/tools/contributors.d
@@ -78,23 +78,39 @@ auto findAuthors(string revRange, FindConfig config)
             enforce(wait(p.pid) == 0, "Failed to execute '%(%s %)'.".format(cmd));
         }
 
+        void runCmd(string[] cmd)
+        {
+            auto p = pipeProcess(cmd, Redirect.stdout);
+            scope(exit) enforce(wait(p.pid) == 0, "Failed to execute '%(%s %)'.".format(cmd));
+
+            authors ~= p.stdout
+                .byLineCopy
+                .tee!(_ => commits++)
+                .map!((line){
+                    auto ps = line.splitter("|");
+                    return GitAuthor(ps.front, ps.dropOne.front);
+                })
+                .filter!(a => a.name != "The Dlang Bot");
+        }
+
         auto cmd = ["git", "-c", "mailmap.file=%s".format(config.mailmapFile), "-C", repo, "log", "--use-mailmap", "--pretty=format:%aN|%aE"];
         if (!config.showAllContributors)
-            cmd ~= revRange;
+        {
+            if (repo.endsWith("/druntime"))
+                cmd ~= "v2.100.0..upstream/stable";
+            else
+                cmd ~= revRange;
+        }
         if (config.noMerges)
             cmd ~= "--no-merges";
-
-        auto p = pipeProcess(cmd, Redirect.stdout);
-        scope(exit) enforce(wait(p.pid) == 0, "Failed to execute '%(%s %)'.".format(cmd));
-
-        authors ~= p.stdout
-            .byLineCopy
-            .tee!(_ => commits++)
-            .map!((line){
-                auto ps = line.splitter("|");
-                return GitAuthor(ps.front, ps.dropOne.front);
-            })
-            .filter!(a => a.name != "The Dlang Bot");
+        if (repo.endsWith("/dmd"))
+        {
+            repo.buildPath("druntime").mkdirRecurse;
+            runCmd(cmd ~ ":!druntime");
+            runCmd(cmd ~ "druntime");
+        }
+        else
+            runCmd(cmd);
     }
     if (!config.showAllContributors)
         stderr.writefln("Looked at %d commits in %s", commits, revRange);
diff --git a/dlang.org/js/listanchors.js b/dlang.org/js/listanchors.js
index b7c113a3..34e7c4f5 100644
--- a/dlang.org/js/listanchors.js
+++ b/dlang.org/js/listanchors.js
@@ -62,45 +62,43 @@ function addVersionSelector() {
       return {
         name: e,
         url: "https://docarchives.dlang.io/v" + e + ".0/" + currentRoot + ddocModuleURL,
-        selected: false,
       };
   });
 
-  var rootURL = location.href.split(/\/(phobos|library|spec)(-prerelease)?/)[0]
   var onlineVersions;
   if (isSpec) {
     onlineVersions = [{
       name: "master",
-      url: rootURL + "/spec/" + ddocModuleURL,
+      url: "https://dlang.org/spec/" + ddocModuleURL,
     }];
   } else {
     onlineVersions = [{
       name: "master",
-      url: rootURL + "/phobos-prerelease/" + ddocModuleURL,
+      url: "https://dlang.org/phobos-prerelease/" + ddocModuleURL,
     },{
       name: "master (ddox)",
-      url: rootURL + "/library-prerelease/" + ddoxModuleURL,
+      url: "https://dlang.org/library-prerelease/" + ddoxModuleURL,
     },{
       name: "stable",
-      url: rootURL + "/phobos/" + ddocModuleURL,
+      url: "https://dlang.org/phobos/" + ddocModuleURL,
     },{
       name: "stable (ddox)",
-      url: rootURL + "/library/" + ddoxModuleURL,
+      url: "https://dlang.org/library/" + ddoxModuleURL,
     }];
   }
 
   // set the current URL as selected
   var currentURL = location.href.split(/[#?]/)[0];
-  onlineVersions.forEach(function(v, i) {
-    onlineVersions[i].selected = v.url === currentURL;
+  var versions = onlineVersions.concat(archivedVersions);
+  versions.forEach(function(v, i) {
+    versions[i].selected = v.url === currentURL;
   });
   // Don't show the option chooser if the page hasn't been recognized
   // For example, Ddox symbol pages are currently not supported
-  if (onlineVersions.filter(function(v){return v.selected}).length === 0)
+  if (versions.filter(function(v){return v.selected}).length === 0)
     return;
 
   // build select box of all versions and append to current DOM
-  var versions = onlineVersions.concat(archivedVersions);
   var options = versions.map(function(e, i){
     return "<option value='" + i + "'" + (e.selected ? "selected" : "") + ">" + e.name + "</option>";
   });
